# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/EGLSyncImpl.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/renderer/EGLSyncImpl.h
--- a/gfx/angle/checkout/src/libANGLE/renderer/EGLSyncImpl.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/EGLSyncImpl.h	Sat Sep 07 13:38:36 2019 +0000
@@ -20,6 +20,11 @@
 class Display;
 }  // namespace egl
 
+namespace gl
+{
+class Context;
+}  // namespace gl
+
 namespace rx
 {
 class EGLSyncImpl : angle::NonCopyable
@@ -30,13 +35,20 @@
 
     virtual void onDestroy(const egl::Display *display) {}
 
-    virtual egl::Error initialize(const egl::Display *display, EGLenum type)     = 0;
+    virtual egl::Error initialize(const egl::Display *display,
+                                  const gl::Context *context,
+                                  EGLenum type)                                  = 0;
     virtual egl::Error clientWait(const egl::Display *display,
+                                  const gl::Context *context,
                                   EGLint flags,
                                   EGLTime timeout,
                                   EGLint *outResult)                             = 0;
-    virtual egl::Error serverWait(const egl::Display *display, EGLint flags)     = 0;
+    virtual egl::Error serverWait(const egl::Display *display,
+                                  const gl::Context *context,
+                                  EGLint flags)                                  = 0;
     virtual egl::Error getStatus(const egl::Display *display, EGLint *outStatus) = 0;
+
+    virtual egl::Error dupNativeFenceFD(const egl::Display *display, EGLint *result) const = 0;
 };
 }  // namespace rx
 